libsplinter/src/circuit/template/mod.rs

```diff
@@ -78,34 +78,33 @@ impl CircuitTemplateManager {
 
     pub fn list_available_templates(&self) -> Result<Vec<String>, CircuitTemplateError> {
         let path = Path::new(&self.path);
-
-        path.read_dir()
-            .map_err(|err| {
-                CircuitTemplateError::new_with_source(
-                    &format!("Failed to read files in {}", self.path),
-                    Box::new(err),
-                )
-            })?
-            .map(|entry| {
-                let file = entry.map_err(|err| {
-                    CircuitTemplateError::new_with_source(
-                        &format!("Failed to read file in {}", self.path),
-                        Box::new(err),
-                    )
-                })?;
-                Ok(file
-                    .file_name()
-                    .into_string()
-                    .map_err(|_| {
-                        CircuitTemplateError::new(&format!(
-                            "Failed to read file name {}",
-                            self.path
-                        ))
-                    })?
-                    .trim_end_matches(".yaml")
-                    .to_string())
-            })
-            .collect::<Result<_, CircuitTemplateError>>()
+        let mut available_templates: Vec<String> = vec![];
```

Comment:
not a necessary change, but this could be more be done idiomatically with `filter_map`.

Reply:
Also good point, changed it to `filter_map`